### DISCUSSION TITLE
[server] 2주차 - 기능정리

### DISCUSSION THREAD

**Discussion by 2020147542** [BookMoA/server#8] [Ideas]
#### 각자 맡은 부분에 대한 기능을 정리해봅시다~

예시는 아래와 같습니다!

**로그인**

- 고객은 아이디, 비밀번호를 입력할 수 있다.
- 고객은 로그인을 할 수 있다.
- 고객은 비밀번호를 찾을 수 있다(심화)
- 고객은 개인정보를 조회할 수 있다(아이디, 닉네임 등).
- 고객별로 등급이 존재하여 볼 수 있는 컨텐츠 상이(심화)
- 고객은 로그아웃을 할 수 있다.

**Comment by 2020147542** [comment]
## 설정+로그인

### 1. 일반 회원가입
- 고객은 이메일, 비밀번호, 확인용 비밀번호, 닉네임을 입력할 수 있다.
- 고객은 반드시 영문과 숫자로 조합된 비밀번호를 입력할 수 있다.(특수문자는 선택)
- 고객은 한글/영어로 이루어진 닉네임을 입력할 수 있다.(특수문자 및 이모티콘 불가)
- 고객은 중복되지 않은 닉네임만 입력할 수 있다.
- 고객은 회원가입을 할 수 있다.

### 2. 회원탈퇴
- 일반 회원가입/소셜 로그인한 고객은 회원탈퇴를 할 수 있다.

### 3. 로그인
- 일반 회원가입한 고객은 아이디, 비밀번호를 입력할 수 있다.
- 일반 회원가입한 고객은 일반 로그인을 할 수 있다.
- 일반 회원가입한 고객은 비밀번호를 찾을 수 있다
- 고객은 카카오 소셜 로그인을 할 수 있다.(닉네임 겹침 예방 필요)

### 4. 로그아웃
- 로그인한 고객은 로그아웃을 할 수 있다.

### 5. 프로필 변경
- 일반 로그인한 고객은 닉네임, 프로필 사진, 이메일을 변경할 수 있다.
- 소셜 로그인한 고객은 변경 가능한지 추후 체크 필요

### 6. 알림 설정
- 로그인한 고객은 푸시 알림을 변경할 수 있다.
- 댓글 푸시, 공감 푸시, 야간 푸시 알람이 존재한다.

### 7. 책모아 팀 소개
- 로그인한 고객은 책 모아 팀원 정보를 확인할 수 있다.

**Reply by 2020147542** [reply]
질문사항
- 비밀번호 찾기 페이지는 없던데 어떻게 진행되는지
- 회원 탈퇴 페이지는 없던데 어떻게 진행되는지
- 회원가입 후 바로 로그인되는지 여부

**Reply by 2020147542** [reply]
(7/21) 카카오 소셜 로그인 프로필 변경 가능 확인
질문사항 피그마에 물어본 상태
모바일도 쿠키 사용이 가능한지 확인(아니면 session으로 변경해야하나,,,?)

**Comment by jinnieusLab** [comment]
## 메인 홈+기록

### 1. 독서 통장
- 고객이 책을 읽고 정보를 기록한다면, 책별로 통장에 그 날 읽은 만큼의 페이지 수가 적립(업데이트)된다.
- 그 날 읽은 만큼의 페이지 수 계산 방법: 독서 메모에서 마지막으로 읽었다고 기록한 페이지와 가장 최근 읽은 페이지로 저장된 값(독서량, 최초로 읽은 책의 기본값은 0)을 뺀 값이 독서 통장으로 적립된다. 
- 고객의 독서 통장에 페이지가 적립될 때 해당 책 제목과 지은이 정보, 그 날 해당 책 읽은 페이지 수를 각 적립 정보로 한다.
- 이때까지 읽은 총 누적 페이지를 확인할 수 있다.

### 2. 독서 집중 모드
- 고객은 원하는 책을 선택하여 독서 집중 모드로 들어갈 수 있다.

### 3. 독서 통계
- 전체 고객들의 독서 통장 내 총 누적 페이지를 기준으로, 고객 본인이 상위 몇 퍼센트에 해당하는지 확인할 수 있다.

### 4. 독서 메모
- 독서 집중 모드 해제 후 독서 메모 화면에서 마지막으로 읽은 페이지를 기록할 수 있다.
- 책 읽은 기간을 기록 및 수정할 수 있다.
- 책 읽은 페이지를 수정할 수 있다.
- 원하는 페이지를 설정한 후 그 곳에 고객이 적고 싶은 내용을 메모 형태로 남길 수 있다.

**Reply by jinnieusLab** [reply]
질문사항
- 독서 메모에서 마지막으로 읽은 페이지를 기록 및 수정할 때, 그 전에 읽었던 페이지보다 작은 수를 입력하면 어떻게 처리할 것인지
- 그 전에 읽었던 페이지보다 작은 수를 입력했을 때 독서 통장 내역은 어떻게 업데이트되는지(독서량이 음수로 계산되니 출금되는 느낌으로? 혹은 이전 통장 내역을 수정?)
- 서재에서 내 보관함>해당 책>독서 메모로 넘어가지는 것인지

**Comment by ldayun** [comment]
## 서재

### 1. 검색
- 고객은 책, 책 리스트 모두 검색을 할 수 있다.
- 고객은 전체, 제목, 지은이, 북리스트, 메모 별로 검색 할 수 있다.
- 고객은 보관함에 있는 책만 검색할 수 있다.

### 2. 리스트
- 고객은 보관함(전체, 읽은 책, 읽고 있는 책, 읽고 싶은 책)을 확인할 수 있다.
- 고객은 리스트를 추가, 수정(이미지, 리스트 제목, 소개글, 공개/비공개)할 수 있다.
- 고객은 리스트를 삭제할 수 있다.
- 고객은 리스트의 책을 추가,삭제할 수 있다.
- 고객은 리스트의 책 순서를 수정할 수 있다.
- 고객은 인기 리스트(좋아요 순) 10개를 볼 수 있다.
- 고객은 책을 정렬할 수 있다.(최신순, 오랜된순, 정확도순, 별점 높은 순, 별점 낮은 순)

### 3. 좋아요
- 고객은 마음에 드는 리스트를 좋아요 할 수 있다.

### 4. 추천 
- 고객은 총 5권을 추천받는다. 가장 인기있는 책 3권 + 랜덤 책 2권을 교차로 넣는다.

**Reply by ldayun** [reply]
질문사항
- 단순 도서 검색은 서버의 책으로만 가져올건지..?  프론트에서 api사용해 받아올거라면 도서 검색은 서버 담당인지? 프론트 담당인지?
- 추천에서 가장 인기 있는 책 3권은 어떻게 정하는지? 리스트는 좋아요순이지만 책도 좋아요가 있는지? 아니면 사람들이 많이 읽은 순?
-  리스트의 책 순서는 프론트에서 저장하는지? 서버에서 저장한다면 테이블에 추가로 필드명이 필요한지?

**Comment by dasol729** [comment]
## 독서 모임
### 1. 독서모임 생성
- 고객은 독서 모임을 새로 생성할 수 있다. (단, 현재 가입된 독서 모임이 1개 이하)

### 2. 독서모임 가입 & 탈퇴
- 고객은 1개의 독서 모임에 가입할 수 있다.
- 고객은 소속된 독서 모임에서 탈퇴할 수 있다. 

### 3. 독서모임 조회
- 고객은 현재 속한 독서 모임을 확인 할 수 있다.
- 고객은 신규, 활동, 마감 임박 등의 기준으로 독서 모임을 조회할 수 있다. 
- 고객은 독서 모임을 검색할 수 있다. (모임명, 공지사항)

### 4. 독서모임 멤버
- 고객은 소속된 독서 모임의 멤버를 조회할 수 있다. 
- 고객은 소속된 독서 모임 멤버들의 상태메세지를 조회할 수 있다. (심화)
- 고객은 소속된 독서 모임에서 자신의 상태메세지를 설정할 수 있다. (심화)

### 5. 독서모임 게시글
- 고객은 소속된 독서 모임에서 게시글을 작성할 수 있다. 
- 고객은 소속된 독서 모임의 게시글을 검색할 수 있다. (글+댓글, 제목, 작성자) 
- 고객은 피드를 통해 소속된 독서 모임의 게시글을 조회할 수 있다. 
- 고객은 소속된 독서 모임에서 게시글에 공감할 수 있다. 
- 고객은 소속된 독서 모임에서 게시글에 공감을 취소할 수 있다. 
- 고객은 소속된 독서 모임에서 게시글에 댓글을 작성할 수 있다.

**Reply by dasol729** [reply]
### 질문사항
- 독서모임 삭제 기능의 존재 여부
- 모임장이 탈퇴하면..?
- 독서모임 수정 (공지사항 등) 기능의 존재 여부
- 게시글 수정, 삭제 기능의 존재 여부
- 게시글의 댓글 수정, 삭제 기능의 존재 여부
